### PR TITLE
Add a simple contributing file.

### PR DESCRIPTION
**Pull request by bbockelm**
Mostly, this is a set of links to other, existing documentation.  However, the GitHub web UI will recommend developers review this file before submitting a pull-request.  It's a simple nudge to "do the right thing".

### CODE REVIEW

**Comment by martinmine** [comment]
What about adding some text that says how clang-format and Jenkins works when contributors submit a PR?

**Comment by bbockelm** [comment]
@martinmine - do you know of any pre-existing text I could link to?

(If not, I could come up with something)

**Comment by martinmine** [comment]
@bbockelm the only thing I can come up with is the how-to for using @phsft-bot: https://github.com/phsft-bot/build-configuration/blob/master/README.md

**Comment by bbockelm** [comment]
Ok, added!

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+While there are several levels of automated tests that will be triggered before a PR is integrated into `master`, we
+strongly encourage new developers to run the tests _before_ submitting new code.  [Please follow the testing docs](https://root.cern.ch/run-tests)
+for more information.
+
+Continuous Integration
+----------------------
+
+ROOT currently has two automated CI tests that are used for pull requests:
+- *Build and test*: When known contributors submit a PR, a Jenkins-based CI workflow will kick off; the results will be posted to the ticket.  Interacting with the Jenkins instance is [documented here](https://github.com/phsft-bot/build-configuration/blob/master/README.md).  The build and tests done for each PR are a subset of those done for nightly builds.
```
Comment (martinmine) [inline CONTRIBUTING.md]: > will be posted to the ticket.
Perhaps "will be posted on the pull request" would be more clearer?

Comment (martinmine) [inline CONTRIBUTING.md]: You should also emphasize that regular contributors/root-project members are the only ones who can interact with the bot and start building PRs.

**Comment by martinmine** [comment]
It looks like I started a review by accident, those comments were general comments, not a part of a review. Oops.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+- *Formatting check*: We utilize a Travis-CI-based build to automatically check that a PR follows known ROOT coding conventions.  This is done with the `clang-format` tool.  If coding violations are found, `clang-format` will generate a suggested patch file; we suggest utilizing this output directly
+
```
Comment (vgvassilev) [inline CONTRIBUTING.md]: @bbockelm, we also have enabled a travis-based `clang-tidy` checks for static analysis issues.

**Comment by bbockelm** [comment]
@martinmine - Since you already started a review, might as well give it an "approve" :)

@vgvassilev - I added comments about `clang-tidy`; I marked the branch as editable by maintainers if you'd like to play with the text further.  Given I can't find any authoritative document on copyright text, I don't want to make one up on the spot here.  Let's get this page done -- we can always return to it later!

**Comment by peremato** [comment]
We do need still to update the documentation pointed by this contributing file, for example how to create a PR.

**Pull request merged**